docs/prod-maintain-secure-upgrade.md

```diff
@@ -1,83 +1,116 @@
+release tarball from
+[https://www.zulip.com/dist/releases/](https://www.zulip.com/dist/releases/)
+
+You also have the option of creating your own release tarballs from a copy of
+zulip.git repository using `tools/build-release-tarball`. And, starting with
+Zulip version 1.4, you can deploy updates [directly from a Git
+repository](#deploying-zulip-from-a-git-repository).
+
+Next, run as root:
+
+```
+/home/zulip/deployments/current/scripts/upgrade-zulip zulip-server-VERSION.tar.gz
+```
+
+The upgrade process will shut down the Zulip service and then run `apt-get upgrade`, a
+puppet apply, any database migrations, and then bring the Zulip service back
+up. Upgrading will result in some brief downtime for the service, which should be
+under 30 seconds unless there is an expensive transition involved. Unless you
+have tested the upgrade in advance, we recommend doing upgrades at off hours.
+
+### Preserving local changes to configuration files
+
+**Warning**: If you have modified configuration files installed by
+Zulip (e.g. the nginx configuration), the Zulip upgrade process will
+overwrite your configuration when it does the `puppet apply`.
+
+You can test whether this will happen assuming no upstream changes to
+the configuration using `scripts/zulip-puppet-apply` (without the
+`-f` option), which will do a test puppet run and output and changes
+it would make. Using this list, you can save a copy of any files
+that you've modified, do the upgrade, and then restore your
+configuration.
+
+If you need to do this, please report the issue so
+that we can make the Zulip puppet configuration flexible enough to
+handle your setup.
+
+### Upgrade log location
+
+The Zulip upgrade script automatically logs output to
+`/var/log/zulip/upgrade.log`. Please use those logs to include output that
+shows all errors in any bug reports.
+
+### Rolling back to a prior version
+
+The Zulip upgrade process works by creating a new deployment under
+`/home/zulip/deployments/` containing a complete copy of the Zulip server code,
+and then moving the symlinks at `/home/zulip/deployments/current` and
+`/root/zulip` as part of the upgrade process.
+
+This means that if the new version isn't working,
+you can quickly downgrade to the old version by using
+`/home/zulip/deployments/<date>/scripts/restart-server` to return to
+a previous version that you've deployed (the version is specified
+via the path to the copy of `restart-server` you call).
+
+### Update settings
+
+If required, you can update your settings by editing `/etc/zulip/settings.py`
+and then run `/home/zulip/deployments/current/scripts/restart-server` to
+restart the server.
+
+### Ubuntu system updates
+
+While the Zulip upgrade script runs `apt-get upgrade`, you are responsible for
+running this on your system on a regular basis between Zulip upgrades to
+ensure that it is up to date with the latest security patches.
+
+### API and your Zulip URL
```

Comment:
@timabbott Is this page the best place for details about the Zulip API url? I think either the _Installing_ or _Customizing Zulip_ pages would be better.